pandas/tests/io/formats/test_to_csv.py

```diff
@@ -223,3 +224,46 @@ def test_to_csv_multi_index(self):
 
         exp = "foo\nbar\n1\n"
         assert df.to_csv(index=False) == exp
+
+    def test_to_csv_compression(self):
+        import gzip
+        import bz2
+
+        df = DataFrame([1])
+        exp = "".join((",0", os.linesep, "0,1", os.linesep)).encode("ascii")
+
+        with tm.ensure_clean("test.gz") as path:
+            df.to_csv(path, compression="gzip")
+            with gzip.GzipFile(path) as f:
+                assert f.read() == exp
+
+        with tm.ensure_clean("test.gz") as path:
+            df.to_csv(path, compression="infer")
+            with gzip.GzipFile(path) as f:
+                assert f.read() == exp
+
+        with tm.ensure_clean("test.bz2") as path:
+            df.to_csv(path, compression="bz2")
+            with bz2.BZ2File(path) as f:
+                assert f.read() == exp
+
+        with tm.ensure_clean("test.bz2") as path:
+            df.to_csv(path, compression="infer")
+            with bz2.BZ2File(path) as f:
+                assert f.read() == exp
+
+    def test_to_csv_compression_lzma(self):
```

Comment:
Reference the issue that you mention in the discussion under the function definition.